compiler/generator_rust.c

```diff
@@ -0,0 +1,1355 @@
+
+#include <stdlib.h> /* for exit */
+#include <string.h> /* for strlen */
+#include <stdio.h> /* for fprintf etc */
+#include "header.h"
+
+/* prototypes */
+
+static void generate(struct generator * g, struct node * p);
+static void w(struct generator * g, const char * s);
+static void writef(struct generator * g, const char * s, struct node * p);
+
+
+enum special_labels {
+    x_return = -1
+};
+
+static int new_label(struct generator * g) {
+
+    int next_label = g->next_label++;
+    g->max_label = (next_label > g->max_label) ? next_label : g->max_label;
```

Comment:
You don't seem to ever use `max_label` (it's needed for Python because there `goto` is emulated by throwing exceptions of different classes and we need to know how many different classes are used so we can declare them).  So you can probably just `return g->next_label++;` here.